contracts/contracts/RecoveryWithDelayPlugin.sol

```diff
@@ -0,0 +1,281 @@
+        );
+
+    struct Announcement {
+        uint64 executionTime; // Block time in seconds when the announced transaction can be executed
+        uint64 validityDuration; // Duration in seconds the announcement is valid after delay is over (0 is valid forever)
+        bool executed; // Flag if the announced transaction was executed
+    }
+
+    // Contract storage
+
+    // Only recoverer can initiate recovery process
+    address public immutable recoverer;
+
+    // Transaction Hash -> Announcement
+    mapping(bytes32 => Announcement) public announcements;
+
+    // Events
+    event NewRecoveryAnnouncement(address account, bytes32 txHash);
+    event RecoveryAnnouncementCancelled(bytes32 txHash);
+    event OwnerReplaced(address account, address oldowner, address newOwner);
+
+    // Errors
+    error CallerNotValidRecoverer();
+    error NonceAlreadyUsed(uint256 nonce);
+    error TransactionAlreadyExecuted(bytes32 txHash);
+    error TransactionAlreadyScheduled(bytes32 txHash);
+    error ExecutionTimeShouldBeInFuture();
+    error TransactionNotFound(bytes32 txHash);
+    error TransactionExecutionNotAllowedYet(bytes32 txHash);
+    error TransactionExecutionValidityExpired(bytes32 txHash);
+
+    constructor(
+        address _recoverer
+    )
+        BasePluginWithEventMetadata(
+            PluginMetadata({name: "Recovery Plugin", version: "1.0.0", requiresRootAccess: true, iconUrl: "", appUrl: ""})
+        )
+    {
+        recoverer = _recoverer;
+    }
+
+    modifier onlyRecoverer() {
+        if (msg.sender != recoverer) {
+            revert CallerNotValidRecoverer();
+        }
+        _;
+    }
+
+    /**
+     * @notice Executes a Safe transaction that swaps owner with a new owner. This allows a Safe account to be recovered
+     *         if the owner's private key is lost. A safe account must set manager as a Module on a safe and enable this
+     *         contract as Plugin on a Safe.
+     * @param manager Address of the Safe{Core} Protocol Manager.
+     * @param safe Safe account whose owner has to be recovered
+     * @param prevOwner Owner that pointed to the owner to be replaced in the linked list
+     * @param oldOwner Owner address to be replaced.
+     * @param newOwner New owner address.
+     * @param nonce A unique identifier used to uniquely identify a recovery transaction.
+     * @return data Bytes returned from the manager contract.
+     */
+    function executeFromPlugin(
+        ISafeProtocolManager manager,
+        ISafe safe,
+        address prevOwner,
+        address oldOwner,
+        address newOwner,
+        uint256 nonce
+    ) external returns (bytes memory data) {
+        bytes32 txHash = getTransactionHash(address(manager), address(safe), prevOwner, oldOwner, newOwner, nonce);
+        Announcement memory announcement = announcements[txHash];
```

Comment:
Does this actually change the storage? on line `110` you set `executed` to true

Reply:
The code sets the `storage` at line `110` (now `109`)